File: types/world-hub.ts
export interface FlightMarker {
  id: string;
  callsign: string;
  latitude: number;
  longitude: number;
  altitude: number;
  speed: number;
  heading: number;
  origin?: string;
  destination?: string;
  airline?: string;
  aircraftType?: string;
  onGround: boolean;
}

export interface ShipMarker {
  id: string;
  name: string;
  latitude: number;
  longitude: number;
  speed: number;
  heading: number;
  shipType: string;
  flag?: string;
  destination?: string;
  status?: string;
  mmsi?: string;
}

export interface AssetSensitivity {
  assetClass: string;
  score: number;
  tickers: string[];
}

export interface GeopoliticalMarker {
  id: string;
  title: string;
  latitude: number;
  longitude: number;
  region: string;
  eventType: string;
  severity: number;
  marketImpact: number;
  affectedAssets: AssetSensitivity[];
  summary: string;
  timestamp: string;
  source: string;
}

export interface WorldHubOverview {
  flightCount: number;
  shipCount: number;
  activeEvents: number;
  globalRiskScore: number;
  topEvents: GeopoliticalMarker[];
  assetClassSensitivity: AssetSensitivity[];
}

export interface FlightsResponse {
  flights: FlightMarker[];
  count: number;
  source: string;
}

export interface ShipsResponse {
  ships: ShipMarker[];
  count: number;
  source: string;
}

export interface GeopoliticalResponse {
  events: GeopoliticalMarker[];
  count: number;
  globalRiskScore: number;
  source: string;
}

export interface LayerToggleState {
  flights: boolean;
  ships: boolean;
  geopolitical: boolean;
  newsSentiment: boolean;
  chokepoints: boolean;
  energyRoutes: boolean;
}

File: backend/app/schemas/world_hub.py
from __future__ import annotations

from pydantic import BaseModel
from typing import Optional


class FlightMarker(BaseModel):
    id: str
    callsign: str
    latitude: float
    longitude: float
    altitude: float
    speed: float
    heading: float
    origin: Optional[str] = None
    destination: Optional[str] = None
    airline: Optional[str] = None
    aircraftType: Optional[str] = None
    onGround: bool = False


class ShipMarker(BaseModel):
    id: str
    name: str
    latitude: float
    longitude: float
    speed: float
    heading: float
    shipType: str
    flag: Optional[str] = None
    destination: Optional[str] = None
    status: Optional[str] = None
    mmsi: Optional[str] = None


class AssetSensitivity(BaseModel):
    assetClass: str
    score: float
    tickers: list[str]


class GeopoliticalMarker(BaseModel):
    id: str
    title: str
    latitude: float
    longitude: float
    region: str
    eventType: str
    severity: float
    marketImpact: float
    affectedAssets: list[AssetSensitivity]
    summary: str
    timestamp: str
    source: str


class WorldHubOverview(BaseModel):
    flightCount: int
    shipCount: int
    activeEvents: int
    globalRiskScore: float
    topEvents: list[GeopoliticalMarker]
    assetClassSensitivity: list[AssetSensitivity]


class FlightsResponse(BaseModel):
    flights: list[FlightMarker]
    count: int
    source: str


class ShipsResponse(BaseModel):
    ships: list[ShipMarker]
    count: int
    source: str


class GeopoliticalResponse(BaseModel):
    events: list[GeopoliticalMarker]
    count: int
    globalRiskScore: float
    source: str
